Lib/test/test_typing.py

```diff
@@ -8606,5 +8606,75 @@ def test_is_not_instance_of_iterable(self):
             self.assertNotIsInstance(type_to_test, collections.abc.Iterable)
 
 
+class TestGenericAliasLike(BaseTestCase):
+    def test(self):
+        T = TypeVar('T')
+
+        class GenericType(Generic[T]):
+            def __class_getitem__(cls, args):
+                # The goal is that the thing returned from here can
+                # act like a GenericAlias but also implement custom behavior
+                # In particular the original feature request involved tracking
+                # TypeVar substitutions at runtime for Pydantic
+                if not isinstance(args, tuple):
+                    args = (args,)
+                parameters = tuple([a for a in args if isinstance(a, TypeVar)])
+                args = tuple([a for a in args if a not in parameters])
+                return GenericAliasLike(args, parameters, cls)
+
+
+        class GenericAliasLike:
+            def __init__(self, args, parameters, origin):
+                self.__args__ = args
+                self.__parameters__ = parameters
+                self.__origin__ = origin
+
+            def __getitem__(slf, args):
+                # Here would go logic to do the tracking of type var substitution
+                # For the purposes of our tests this hardcoded version does just fine
+                self.assertEqual(args, (int,))
+                slf.__args__ = args
+                slf.__parameters__ = ()
+                return slf
+
+            def __eq__(self, other):
+                # implemented just for easy comparison in the tests
+                # below
+                if not isinstance(other, GenericAliasLike):
+                    return False
+                return (
+                    self.__args__ == other.__args__
+                    and
+                    self.__origin__ == other.__origin__
+                    and
+                    self.__parameters__ == other.__parameters__
```

Comment:
oops, more PEP8, sorry

```suggestion
                    and self.__origin__ == other.__origin__
                    and self.__parameters__ == other.__parameters__
```

Reply:
Don't apologize, you're helping me out, I'm just used to living in the "if it doesn't get autoformatted I don't know what to do" world.